publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -311,37 +296,51 @@ public ResponseEntity generateAndPublish(final String msgProtocol, final String
-                        String validResponseBody = validResponse.toString();
-                        SendResult result = messageService.send(validResponseBody, msgService, userDomain, tag, routingKey);
-                        eventResponse.put(JSON_STATUS_RESULT, result);
-                    }
-                } else {
-                    eventResponse.put(JSON_EVENT_MESSAGE_FIELD, JsonResponseEvent);
-                }
-                responseEvents.add(eventResponse);
-            }
+            responseEvents = processingValidEvent(e.getResponseBodyAsString(), msgProtocol, userDomain,
+                    tag, routingKey);
             return new ResponseEntity<>(responseEvents, HttpStatus.BAD_REQUEST);
         }
         return new ResponseEntity<>(responseEvents, HttpStatus.OK);
     }
 
+    public List<Map<String, Object>> processingValidEvent(String eventMsg, final String msgProtocol, final String userDomain,
+                                                          final String tag, final String routingKey) {
+        MsgService msgService = PublishUtils.getMessageService(msgProtocol, msgServices);
+        List<Map<String, Object>> responseEvent = new ArrayList<>();
+        Map<String, Object> eventResponse;
+        JsonElement element = JsonParser.parseString(eventMsg);
+        for (int i = 0; i < element.getAsJsonArray().size(); i++) {
+            eventResponse = new HashMap<>();
+            JsonElement jsonResponseEvent = element.getAsJsonArray().get(i);
+            if (isValid(jsonResponseEvent)) {
+                synchronized (this) {
+                    JsonObject validResponse = jsonResponseEvent.getAsJsonObject();
+                    String validResponseBody = validResponse.toString();
+                    SendResult result = messageService.send(validResponseBody, msgService, userDomain, tag, routingKey);
+                    eventResponse.put(JSON_STATUS_RESULT, result);
+                }
+            } else {
+                eventResponse.put(JSON_EVENT_MESSAGE_FIELD, jsonResponseEvent);
+            }
+            responseEvent.add(eventResponse);
+        }
+        return responseEvent;
+    }
+
+    public JsonObject getAsJsonObject(JsonElement jsonElement) {
+        JsonObject jsonObject = jsonElement.getAsJsonObject();
+        return jsonObject;
+    }
+
     /**
      * This helper method check the json event is valid or not
      */
 
-    private  Boolean isValid(JsonElement jsonObject) {
+    private  Boolean isValid(JsonElement jsonElement) {
         try {
-            return jsonObject.getAsJsonObject().has("meta") && jsonObject.getAsJsonObject().has("data") &&
-                    jsonObject.getAsJsonObject().has("links") && !jsonObject.getAsJsonObject().has("Status code");
-        } catch (Exception e) {
+            return getAsJsonObject(jsonElement).has(META) && getAsJsonObject(jsonElement).has("data") &&
```

Comment:
Use constants instead of hard-coded values, i.e. "data" and "links". There're constants defined in Event class or somewhere...